qdao/circuit.py

```diff
@@ -109,7 +112,99 @@ def run(self, circuit: Any) -> List[QdaoCircuit]:
         if instrs:
             sub_circ = self._circ_helper.gen_sub_circ(instrs, self._nl, self._np)
             sub_circs.append(sub_circ)
+        print("----------StaticPartitioner-----------")
+        print("num of sub-circuits:" + str(len(sub_circs)))
+        return sub_circs
+
+
+class dptask:
+    """To assist the Uniq partitioning algorithm"""
+
+    def __init__(self, qubitNum: int, gateNum: int):
+        self.gateNum = gateNum
+        self.qubitNum = qubitNum
+
+    def addGate(self, gateIndex: int, gatePos):
+        if isinstance(gatePos, int):
+            gatePos = [gatePos]
+        for j in range(self.qubitNum):
+            self.result_bit[gateIndex + 1][j] += self.result_bit[gateIndex][j]
+            self.result_op[gateIndex + 1][j] += self.result_op[gateIndex][j]
+            if j in gatePos:
+                self.result_op[gateIndex + 1][j].append(gateIndex + 1)
+                if len(gatePos) == 2:
+                    a = [x for x in gatePos if x != j][0]
+                    self.result_bit[gateIndex + 1][j] += self.result_bit[gateIndex][a]
+                    self.result_bit[gateIndex + 1][j] = list(
+                        set(self.result_bit[gateIndex + 1][j])
+                    )
+                    self.result_op[gateIndex + 1][j] += self.result_op[gateIndex][a]
+                    self.result_op[gateIndex + 1][j] = list(
+                        set(self.result_op[gateIndex + 1][j])
+                    )
+
+    def createTask(self, ops):
```

Comment:
This function name will confuse users and developers. Basically, we already defined `DpTask` class, and one might think that `DpTask()` will create a task object. So I suggest to consider better naming to distinguish this function from vanilla `DpTask` instantiation.